### PR TITLE
Refactor/ml 467 unify ml ops serialisation tests

### PR DESCRIPTION
**Pull request by baykaner**
- moved layers & ops serialisation tests into ml/test/unit/serializers
- helper functions implemented to reduce repeated code
- refactors ops tests to anonymous namespaces
- build times should be dramatically faster now there is only 1 test that includes graph_builder

### CODE REVIEW

File: libs/ml/tests/unit/serializers/serialisation.cpp [hunk 1]
```diff
@@ -0,0 +1,4407 @@
...
+  EXPECT_TRUE(prediction.at(1).AllClose(
+      new_prediction.at(1), fetch::math::function_tolerance<typename TypeParam::Type>(),
+      fetch::math::function_tolerance<typename TypeParam::Type>()));
+}
+
+//////////////////
+/// EMBEDDINGS ///
+//////////////////
+
+TYPED_TEST(SaveParamsTest, embeddings_saveparams_test)
```
Comment (vhavrylov-ncube) [inline libs/ml/tests/unit/serializers/serialisation.cpp]: Do these two tests (for Embeddings I mean) use graph_builder?

File: libs/ml/include/ml/utilities/graph_builder.hpp [hunk 2]
```diff
@@ -261,14 +273,14 @@ void BuildNodeAndInsertTrainables(NodeSaveableParams<T> const &nsp, std::string
     g->AddTrainable(node, name);
     break;
```
Comment (vhavrylov-ncube) [inline libs/ml/include/ml/utilities/graph_builder.hpp]: I wonder if we could split this header file to both cpp and hpp file, and move all Ops includes to cpp file (and implementation of all functions that are declared here). This way we can reduce include amount in this particular header and reduce its impact on unit test .cpp files without merging many UT .cpp files to a large one.

Comment (vhavrylov-ncube) [reply libs/ml/include/ml/utilities/graph_builder.hpp]: this header is also included in vm-modules Model (VMModel) but it actually does not need the whole scope of includes too and could be slightly refactored; the same for vm-modules Graph.

File: libs/ml/tests/unit/serializers/serialisation.cpp (hunk 1 above)
Comment (baykaner) [reply libs/ml/tests/unit/serializers/serialisation.cpp]: doesn't look like it; why are you asking? There are some tests in here that use BuildGraph and some that do not...

**Review by iryna-melnyk — COMMENTED** [review]
looks much better now, let's see how it builds

**Review by iryna-melnyk — APPROVED** [review]

**Review by vhavrylov-ncube — APPROVED** [review]
LGTM

**Pull request merged**